tests/allatonce/test_allatonceform.py

```diff
@@ -91,8 +92,8 @@ def form_mass(u, v):
         unp1 = ufulls[i]
         v = vfulls[i]
         tform = form_mass(unp1 - un, v/dt)
-        tform += theta*form_function(unp1, v) + (1-theta)*form_function(un, v)
-
+        tform += theta*form_function(unp1, v, time) + (1-theta)*form_function(un, v, time-dt)
+        time.assign(time + dt)
```

Comment:
Using the same `Constant` for the time at every timestep means that when you update the value of `time` here, it will change the value at every timestep. There should be an array of `Constant`s, one for the time at each timestep just like there is in the form (but here there should be enough for the whole window, not just the slice).

Reply:
Resolved by 9999e2cfa2f1d82bbc0a3ec4dbcd258cbd0f5db4